app/controllers/comments_controller.rb

```diff
@@ -108,6 +111,13 @@ def check_unreviewed
     redirect_to logged_in? ? root_path : new_user_session_path
   end
 
+  def check_if_commentable_frozen
+    return unless @commentable&.respond_to?(:iced?) && @commentable&.iced?
+
+    flash[:error] = ts("Sorry, you cannot reply to a frozen comment.")
+    redirect_to(request.env["HTTP_REFERER"] || root_path)
```

Comment:
Would it be possible to use `redirect_back(fallback_location: root_path)` here instead?